File: java_solutions/388.longest-absolute-file-path.java
/*
 * @lc app=leetcode id=388 lang=java
 *
 * [388] Longest Absolute File Path
 */

record Solution() {
    public int lengthLongestPath(String input) {
        return (int) java.util.Arrays.stream(input.split("\n")).reduce(new Object[]{new int[input.length() + 2], 0}, (s, p) -> new Object[]{java.util.stream.Stream.of((int[])s[0]).peek(arr -> arr[p.lastIndexOf("\t") + 2] = arr[p.lastIndexOf("\t") + 1] + p.length() - p.lastIndexOf("\t") - (p.lastIndexOf("\t") == -1 ? 0 : 1)).findFirst().get(), p.contains(".") ? Math.max((int)s[1], ((int[])s[0])[p.lastIndexOf("\t") + 1] + p.length() - p.lastIndexOf("\t") - 1) : (int)s[1]}, (a, b) -> a)[1];
    }
}

File: kotlin_solutions/388.longest-absolute-file-path.kt
/*
 * @lc app=leetcode id=388 lang=kotlin
 *
 * [388] Longest Absolute File Path
 */

class Solution {fun lengthLongestPath(input: String): Int = input.split("\n").fold(mutableListOf<Int>() to 0) { (stack, maxLen), line -> line.takeWhile { it == '\t' }.length.let { level -> line.drop(level).let { name -> stack.apply { while (size > level) removeLast() }.let { s -> if (name.contains('.')) s to maxOf(maxLen, s.sum() + name.length + s.size) else s.apply { add(name.length) } to maxLen } } } }.second}